magefiles/magefile.go

```diff
@@ -427,6 +427,10 @@ func (ci CI) setRequiredEnvVars() error {
 				envVarPrefix = "SPI_OPERATOR"
 				imageTagSuffix = "spi-image"
 				testSuiteLabel = "spi-suite"
+
+				// spi also requires service-provider-integration-oauth image
+				os.Setenv("SPI_OAUTH_IMAGE_REPO", os.Getenv("CI_SPI_OAUTH_IMAGE"))
```

Comment:
I think this line won't work.  the value in `$CI_SPI_OAUTH_IMAGE` will contain the full image ref including sha, e.g. 
```
image-registry.openshift-image-registry.svc:5000/ci-op-zgmjgd7c/pipeline@sha256:9733e2e684470f1ef50f2f952302e10d102dc57eb676cfd73da95632fd240892
```
then the image in infra-deployments would be set [here](https://github.com/redhat-appstudio/infra-deployments/blob/e082d77a5bd64d735af35266ac3de35cb6b57412/components/spi/overlays/development/kustomization.yaml#L14-L15) to
```
  - name: quay.io/redhat-appstudio/service-provider-integration-oauth
    newName: image-registry.openshift-image-registry.svc:5000/ci-op-zgmjgd7c/pipeline@sha256:9733e2e684470f1ef50f2f952302e10d102dc57eb676cfd73da95632fd240892
    newTag: redhat-appstudio-spi-oauth-image
```

What I think needs to be done here is to split the value of `$CI_SPI_OAUTH_IMAGE` with "@" (like [here](https://github.com/redhat-appstudio/e2e-tests/pull/876/files#diff-d7a1577b2d546e34aaf816bdd1b83edef28a66a555f049767bafa83268b7efbbR312)) and then use the first part of the image, i.e. `image-registry.openshift-image-registry.svc:5000/ci-op-zgmjgd7c/pipeline`

Reply:
Thanks for the suggestion! Changing that.

Reply:
to be absolutely sure and validate this change I suggest to create a test PR in spi repository (which gets paired with this PR), wdyt?

Reply:
@psturc changed, that with the split value. should be fine now, wdyt?

Reply:
yes, it makes sense. i will try pairing the PRs. Thanks!